### ISSUE TITLE
Sequences of attrs validators

### ISSUE THREAD

**Issue by AdrianSosic** [python/mypy#20013]
**Bug Report**

It's only a minor issue, but there was a recent [fix](https://github.com/python-attrs/attrs/issues/1243) on the attrs side to properly handle lists of validators. While code execution works, mypy is still giving a false positive.

**To Reproduce**

```python
from attrs import define, field
from attrs.validators import deep_iterable, instance_of, min_len


@define
class A:
    x: list[str] = field(
        validator=deep_iterable(
            member_validator=[instance_of(str), min_len(1)],
            iterable_validator=[instance_of(list), min_len(1)],
        ),
    )

A(['a', 'b'])  # works
```

**Expected Behavior**

Should not raise any mypy errors.

**Actual Behavior**

```bash
dbg.py:9: error: Argument "member_validator" to "deep_iterable" has incompatible type "list[function]"; expected "Callable[[Any, Attribute[Never], Never], Any] | Sequence[Callable[[Any, Attribute[Never], Never], Any]]"  [arg-type]
dbg.py:10: error: Argument "iterable_validator" to "deep_iterable" has incompatible type "list[function]"; expected "Callable[[Any, Attribute[Never], Never], Any] | Sequence[Callable[[Any, Attribute[Never], Never], Any]] | None"  [arg-type]
```

**Your Environment**

- Mypy version used: 1.18.2
- Attrs version used: 25.4.0
- Python version used: 3.10

**Comment by A5rocks** [comment]
I think this is an artifact of the fact that mypy joins the entries of the list rather than uses a union. It should be possible to do something like:

```py
my_validator_list: list[Callable[[Any, Attribute[str], str], Any]] = [instance_of(str), min_len(1)]

# the rest of your code, with [instance_of(str), min_len(1)] replaced by `my_validator_list`
```

and similar for `iterable_validator`. (fyi the typing for `deep_iterable` is suboptimal -- the type of `member_validator`'s usage of `_T` is useless)

**Comment by A5rocks** [comment]
... Though arguably this should work because type context? I'm not sure.

**Comment by A5rocks** [comment]
~~https://github.com/python/mypy/pull/20025 fixes the `iterable_validator` error.~~ (never mind, I misread. It doesn't) The other error is just how we join things together, and could be fixed in `attrs`'s stubs by replacing the `_T` in `deep_iterable` with `Any`.
